listen/map-changes.js

```diff
@@ -60,12 +69,15 @@ MapChanges.prototype.addMapChangeListener = function (listener, token, beforeCha
         listeners = descriptor.changeListeners;
     }
     listeners.push(listener);
-    Object.defineProperty(this, "dispatchesMapChanges", {
-        value: true,
-        writable: true,
-        configurable: true,
-        enumerable: false
-    });
+    if(Object.getOwnPropertyDescriptor(this.__proto__,"dispatchesMapChanges") === UNDEFINED) {
```

Comment:
void 0 can be faster than looking up the UNDEFINED symbol in the scope chain, and `== null` is a good idiom for "null or undefined" that also avoids the reference resolution.